src/local/butler/scripts/github_metadata_backfiller.py

```diff
@@ -0,0 +1,105 @@
+from libs.issue_management import oss_fuzz_github
+
+
+def _get_testcase(bug_information):
+  """Retrieve testcase based on its bug information."""
+  candidates = list(
+      data_types.Testcase.query(
+          data_types.Testcase.bug_information == bug_information))
+  if not candidates:
+    print('No candidate found when '
+          'querying bug information {bug_information}\n')
+  elif len(candidates) > 1:
+    print('Multiple candidates found when '
+          f'querying bug information {bug_information}:\n')
+    for testcase in candidates:
+      print(f'  {testcase.key.id()}')
+  else:
+    return candidates[0]
+  return None
+
+
+def _get_bug_information(issue):
+  """Given a GitHub issue, parse its corresponding bug information."""
+  bug_information = issue.title[len(oss_fuzz_github.ISSUE_TITTLE_TEXT_PREFIX) +
+                                1:]
+  if bug_information.isdigit():
+    return bug_information
+  return None
+
+
+def _testcase_information_verified(testcase, issue):
+  """Verify if a testcase correctly stores its GitHub issue information."""
+  if testcase.github_repo_id == issue.repository.id and \
+          testcase.github_issue_num == issue.number:
+    print(f'Testcase {testcase.bug_information} was properly stored.')
+    return True
+
+  if testcase.github_repo_id is None and testcase.github_issue_num is None:
+    print(f'Testcase {testcase.bug_information} was not stored.')
+  else:
+    print(f'Testcase {testcase.bug_information} stored '
+          f'is inconsistent with GitHub:\n'
+          f'  Issue number  (Storage) {testcase.github_issue_num} '
+          f'!= {issue.number} (GitHub)\n'
+          f'  Repository ID (Storage) {testcase.github_repo_id} '
+          f'!= {issue.repository.id} (GitHub).')
+  return False
+
+
+def execute(args):
+  """Backtrack GitHub issues filed in the past,
+    update their information in gcloud, and close them when necessary."""
+  issue_tracker = issue_tracker_utils.get_issue_tracker('oss-fuzz')
+  for issue in oss_fuzz_github.get_my_issues():
+    print('========================================')
+    # Track testcase.
+    bug_information = _get_bug_information(issue)
+    if not bug_information:
+      print('Unable to extract bug information: '
+            f'Repo {issue.repository.id} Issue {issue.number}.\n'
+            f'Issue title: {issue.title}.')
+      continue
+    testcase = _get_testcase(bug_information)
+
+    # Update testcase.
+    if not _testcase_information_verified(testcase, issue):
+      print(
+          f'Updating testcase (bug information: {testcase.bug_information}):\n'
+          f'  Issue number  {issue.number}\n'
+          f'  Repository ID {issue.repository.id}\n')
+      if args.non_dry_run:
+        oss_fuzz_github.update_testcase_properties(testcase, issue.repository,
+                                                   issue)
+        testcase.put()
+
+    # Backclose issues.
+    if issue.state == 'closed':
```

Comment:
We can also filter out closed issues when [querying GitHub](https://github.com/google/clusterfuzz/blob/52737a91285799406726f8bfc64f0d5695b50519/src/appengine/libs/issue_management/oss_fuzz_github.py#L226), each method have its pros and cons.
Doing it here is safer. Although we cannot close an issue if we don't have its info in the cloud, the repo owners might close it manually. I suppose this is trivial because we don't really need an issue's info once the issue is closed.

Doing it in the search query can lower the chance of reaching our [rate limit](https://docs.github.com/en/rest/search#rate-limit). When a query reaches the limit, it returns nothing and this script won't work.

I choose to do it here because:
1. We might need the github info of those issues later.
2. My local experiments never reached our rate-limit.